db/resync_status.go

```diff
@@ -0,0 +1,77 @@
+package db
+
+import (
+	"sync"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+type ResyncManager struct {
+	Status           ResyncStatus
+	LastError        error
+	Mutex            sync.Mutex      // Used to lock the Status and when setting / reading LastError
+	ResyncTerminator base.AtomicBool // Allows resync operation to be cancelled while in progress
```

Comment:
Is an atomic bool still necessary if you've got a Mutex for the Manager?  Seems like interaction with all resync state should be done through the manager now.